oktapam/client/cloud_connection.go

```diff
@@ -0,0 +1,199 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"regexp"
+
+	"github.com/okta/terraform-provider-oktapam/oktapam/constants/attributes"
+	"github.com/okta/terraform-provider-oktapam/oktapam/logging"
+	"github.com/okta/terraform-provider-oktapam/oktapam/utils"
+	"github.com/tomnomnom/linkheader"
+)
+
+type CloudConnection struct {
+	ID                     *string                 `json:"id,omitempty"`
+	Name                   *string                 `json:"name"`
+	Provider               *string                 `json:"provider"`
+	CloudConnectionDetails *CloudConnectionDetails `json:"cloud_connection_details"`
+}
+
+type CloudConnectionDetails struct {
+	AccountId  *string `json:"account_id"`
+	ExternalId *string `json:"external_id"`
+	RoleArn    *string `json:"role_arn"`
+}
+
+type CloudConnectionsListResponse struct {
+	CloudConnections []CloudConnection `json:"list"`
+}
+
+func (c CloudConnection) Exists() bool {
+	return utils.IsNonEmpty(c.ID)
+}
+
+func (c CloudConnection) ToResourceMap() map[string]any {
+	m := make(map[string]any)
+
+	if c.ID != nil {
+		m[attributes.ID] = *c.ID
+	}
+	if c.Name != nil {
+		m[attributes.Name] = *c.Name
+	}
+	if c.Provider != nil {
+		m[attributes.CloudConnectionProvider] = c.Provider
+	}
+	if c.CloudConnectionDetails != nil {
+		flattenedDetails := make([]any, 1)
+		flattenedDetail := make(map[string]any)
+		if c.CloudConnectionDetails.AccountId != nil {
+			flattenedDetail[attributes.CloudConnectionAccountId] = *c.CloudConnectionDetails.AccountId
+		}
+		if c.CloudConnectionDetails.ExternalId != nil {
+			flattenedDetail[attributes.CloudConnectionExternalId] = *c.CloudConnectionDetails.ExternalId
+		}
+		if c.CloudConnectionDetails.RoleArn != nil {
+			flattenedDetail[attributes.CloudConnectionRoleARN] = *c.CloudConnectionDetails.RoleArn
+		}
+		flattenedDetails[0] = flattenedDetail
+		m[attributes.CloudConnectionDetails] = flattenedDetails
+	}
+
+	return m
+}
+
+func validateCloudConnectionData(cloudConnection CloudConnection) bool {
+	nameRegex, nameRegexErr := regexp.Compile(`^[A-Za-z0-9-_.]+$`)
+	if nameRegexErr != nil {
+		fmt.Println("invalid cloud connection name")
+		return false
+	}
+	nameValidation := nameRegex.MatchString(*cloudConnection.Name) && len(*cloudConnection.Name) > 1
+
+	accountIdRegex, accountIdRegexErr := regexp.Compile(`^\d{12}$`)
+	if accountIdRegexErr != nil {
+		fmt.Println("invalid cloud connection account id")
+		return false
+	}
+	accountIdValidation := accountIdRegex.MatchString(*cloudConnection.CloudConnectionDetails.AccountId)
+
+	providerValidation := *cloudConnection.Provider == "aws"
```

Comment:
lets also define any string used in the code as `const` or `var`

Reply:
Done.